paddle/fluid/operators/batch_norm_op.cc

```diff
@@ -554,6 +589,20 @@ class BatchNormGradKernel<platform::CPUDeviceContext, T>
     const DataLayout data_layout =
         framework::StringToDataLayout(data_layout_str);
 
+    // batch_norm with inplace as false will take X as grad input, which
+    // is same as cuDNN batch_norm backward calculation, batch_norm
+    // with inplace as true only take Y as input and X should be calculate
+    // by inverse operation of batch_norm on Y
+    const Tensor *x;
+    bool is_inplace;
+    if (ctx.HasInput("Y")) {
+      x = ctx.Input<Tensor>("Y");
+      is_inplace = true;
```

Comment:
`PADDLE_ENFORCE(dy == dx)` to avoid bug?

Reply:
Done, thanks!